qiskit_experiments/characterization/t1_experiment.py

```diff
@@ -0,0 +1,231 @@
+            # problem for the fitter if one of the std points is
+            # exactly zero
+            if stddevs[i] == 0:
+                stddevs[i] = 1e-4
+
+        if t1_guess is None:
+            t1_guess = np.mean(delays)
+        else:
+            t1_guess = t1_guess * dt_factor_in_microsec
+        if offset_guess is None:
+            offset_guess = means[-1]
+        if amplitude_guess is None:
+            amplitude_guess = means[0] - offset_guess
+        if t1_bounds is None:
+            t1_bounds = [0, np.inf]
+        if amplitude_bounds is None:
+            amplitude_bounds = [0, 1]
+        if offset_bounds is None:
+            offset_bounds = [0, 1]
+
+        fit_out, fit_err, fit_cov, chisq = curve_fit_wrapper(
+            exp_fit_fun,
+            delays,
+            means,
+            stddevs,
+            p0=[amplitude_guess, t1_guess, offset_guess],
+        )
+
+        analysis_result = AnalysisResult(
+            {
+                "value": fit_out[1],
+                "stderr": fit_err[1],
+                "unit": result_unit,
+                "label": "T1",
+                "fit": {
+                    "params": fit_out,
+                    "stderr": fit_err,
+                    "labels": ["amplitude", "T1", "offset"],
+                    "chisq": chisq,
+                    "cov": fit_cov,
+                },
+                "quality": self._fit_quality(fit_out, fit_err, chisq),
+            }
+        )
+
+        return analysis_result, None
+
+    @staticmethod
+    def _fit_quality(fit_out, fit_err, chisq):
+        # pylint: disable = too-many-boolean-expressions
+        if (
+            abs(fit_out[0] - 1.0) < 0.1
+            and abs(fit_out[2]) < 0.1
+            and chisq < 3
+            and (fit_err[0] is None or fit_err[0] < 0.1)
+            and (fit_err[1] is None or fit_err[1] < fit_out[1])
+            and (fit_err[2] is None or fit_err[2] < 0.1)
+        ):
+            return "computer_good"
+        else:
+            return "computer_bad"
+
+
+class T1Experiment(BaseExperiment):
+    """T1 experiment class"""
+
+    __analysis_class__ = T1Analysis
+
+    def __init__(
+        self,
+        qubit: int,
+        delays: Union[List[float], np.array],
+        unit: Optional[str] = "us",
+        experiment_type: Optional[str] = None,
+    ):
+        """
+        Initialize the T1 experiment class
+
+        Args:
+            qubit: the qubit whose T1 is to be estimated
+            delays: delay times of the experiments
+            unit:Optional, unit of the duration. Supported units: 's', 'ms', 'us', 'ns', 'ps', 'dt'.
+            experiment_type: Optional, the experiment type string.
+
+        Raises:
+            ValueError: if the number of delays is smaller than 3
+        """
+        if len(delays) < 3:
+            raise ValueError("T1 experiment: number of delays must be at least 3")
+
+        self._delays = delays
```

Comment:
```suggestion
        self._delays = np.asarray(delays, dtype=float)
```

Reply:
@chriseclectic The `delay` gate doesn't accept `np` types